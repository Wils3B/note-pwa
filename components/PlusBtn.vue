<template>
  <v-btn fixed dark fab color="primary" class="plus-btn" to="/edit-note/new">
    <v-icon>add</v-icon>
  </v-btn>
</template>

<script>
export default {
  name: 'PlusButton'
}
</script>

<style scoped>
.plus-btn {
  bottom: 16px;
  right: 16px;
}
</style>
