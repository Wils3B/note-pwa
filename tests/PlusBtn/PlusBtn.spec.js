import { mount } from '@vue/test-utils'
import Vuetify from 'vuetify'
import { localInstance } from '../common'
import router from '../router'
import PlusBtn from '~/components/PlusBtn.vue'

const localVue = localInstance()

describe('Plus Button', () => {
  const wrapper = mount(PlusBtn, {
    localVue,
    router,
    vuetify: new Vuetify(),
  })

  test('Have a correct render', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('it should have correct icon', () => {
    expect(wrapper.find('i.v-icon').text()).toBe('add')
  })

  test('it should be a link to edit new note', () => {
    expect(wrapper.attributes('href')).toBe('#/edit-note/new')
  })
})
