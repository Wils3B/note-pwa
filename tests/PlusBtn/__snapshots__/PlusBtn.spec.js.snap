// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plus Button Have a correct render 1`] = `"<a href=\\"#/edit-note/new\\" class=\\"plus-btn v-btn v-btn--is-elevated v-btn--fab v-btn--fixed v-btn--has-bg v-btn--round v-btn--router theme--dark v-size--default primary\\"><span class=\\"v-btn__content\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate material-icons theme--dark\\">add</i></span></a>"`;
